### PR TITLE
informational and warning constraint results

### PR DESCRIPTION
**Pull request by wandmagic**
# Committer Notes

make test runner aware of informational constraint results
### All Submissions:

- [ ] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [ ] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Review by aj-stein-gsa — COMMENTED** [review]
Are we going to bump the version of oscal-js in this PR or a follow-on to address other bug fixes (oscal-js#10) we probably want in the target branch?

Fine either way, just thought I should ask now. If you want, I can PR that separately. I set it up and reviewing quickly now.

File: features/steps/fedramp_extensions_steps.ts [hunk 1]
```diff
@@ -303,7 +303,7 @@ async function checkConstraints(
 
       const kinds = constraintResults.map((c) => c.kind);
       const passCount = kinds.filter((k) => k === "pass").length;
-      const failCount = kinds.filter((k) => k === "fail").length;
+      const failCount = kinds.filter((k) => k === "fail"||k==='informational').length;
```
Comment (aj-stein-gsa) [inline features/steps/fedramp_extensions_steps.ts:306]: Minor, non-blocking: do we have some style requirements yet? In this case we are using both quote styles and we might as well "stick with tradition.' (Yes, I did that last one to be funny.)

**Review by aj-stein-gsa — DISMISSED** [review]
I am ok with this for now but I will open some separate issues about the structure of the output in the results of the test runner and what it means. At first glance, the wording was confusing and looking at the test harness helped, but I am still not 100% clear.

As an aside: thanks for this cool new Cucumber setup. First take: I like it. 😄

**Review by aj-stein-gsa — APPROVED** [review]

**Comment by wandmagic** [comment]
@aj-stein-gsa  this needs a re-review because i noticed we need to address warning level constraints

@Gabeblis  can you pull this and use your in progress constraints to confirm this fix works?

File: features/steps/fedramp_extensions_steps.ts [hunk 2]
```diff
@@ -301,9 +301,16 @@ async function checkConstraints(
         continue;
       }
 
-      const kinds = constraintResults.map((c) => c.kind);
+      const kinds = constraintResults.map((c) => {
+        if(c.level==='warning'){
+          return 'fail'
+        }else{
+          return c.kind
+      }});
```
Comment (aj-stein-gsa) [inline features/steps/fedramp_extensions_steps.ts:309]: So maybe I misunderstood something on the call, maybe you can explain here in chat, why are warnings always perceived as `fail`?

Comment (aj-stein-gsa) [reply features/steps/fedramp_extensions_steps.ts:309]: Discussed with Paul, I did not understand the interaction `level` and `kind`.

**Review by aj-stein-gsa — APPROVED** [review]

**Comment by wandmagic** [comment]
> I am ok with this for now but I will open some separate issues about the structure of the output in the results of the test runner and what it means. At first glance, the wording was confusing and looking at the test harness helped, but I am still not 100% clear.
> 
> As an aside: thanks for this cool new Cucumber setup. First take: I like it. 😄

I think it might be good to change the final check from expect "pass" to expect failing case to fail or passing case to pass, so that it is clearly a failing of passing test that failed, also the red output from oscal-cli can be confusing, because if you see red you assume something is off. I would like to address both of these concerns in a future PR

**Pull request merged**